create_layout.py

```diff
@@ -91,45 +91,111 @@
 import in_toto.models.link
 import in_toto.models.layout
 
-def create_material_rules(links, index):
+def changes_between_snapshots(before_dict, after_dict):
+  '''Given two 'snapshots' of a file structure -- 'before' and 'after' -- 
```

Comment:
I think it's worth being ambiguous about whether they're files or not. Here, we don't really care about the type of the artifacts. Let's just go with artifacts throughout?